Assets/Scripts/Models/HealthSystem.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+
+[Serializable]
+public class HealthSystem
+{
+    //all of this data should probably be generated at runtime or loaded from a file
+    public float Value
+    {
+        get
+        {
+            return _currentHealth;
+        }
+
+        private set
+        {
+            if (Invincible == true) return;
+            //We have been Revived!
+            if (_currentHealth <= 0 && value > _currentHealth)
```

Comment:
Imagine 
```csharp
var health = new HealthSystem
health = -5.0f;
// We are now dead
health = -3.0f; // (_currentHealth <= 0 && value > _currentHealth) will be true, so we revive even though our health is still < 0.
```

I think the condition should be `(_currentHealth <= 0 && value > 0)`